File: Scarab-0.1.00d19/perl/time-echo.pl
#! /usr/bin/perl

use lib 'lib';

use RPC::Scarab::Manager;

die "usage: $0 [--xml] URL [COUNT]\n"
    if ($#ARGV < 1);

my @marshal_args = ();
if ($ARGV[0] eq '--xml') {
    @marshal_args = (marshal => 'xml');
    shift;
}
my ($url, $count) = @ARGV;

$host = 'localhost' if !defined $host;

$server = RPC::Scarab::Manager->new_rootproxy(connect_to => {PeerAddr => $url,
							  Proto => 'tcp'},
					   @marshal_args);

$object = "object";

$time_zero = time();
for ($ii = 0; $ii < $count; $ii ++) {
    $server->echo($object);
}
$total_time = time() - $time_zero;
$ms = ($total_time / $count) * 1000;

print "$count calls in $total_time seconds ${ms}ms/call\n";

File: supporting_libs/scarab/Scarab-0.1.00d19/python/time-echo.py
import Scarab
import sys
import time
import string

if len(sys.argv) != 2 and len(sys.argv) != 3:
	raise "usage error", "usage: tim-echo URL [COUNT]"

connection = Scarab.connect_to(sys.argv[1])
if len(sys.argv) == 3:
    count = int(sys.argv[2])
else:
    count = 1000

t1 = time.time()
for ii in range(count):
	connection.call('root', 'echo', ['testing'])
total_time = time.time() - t1
ms = (float(total_time) / float(count)) * 1000.0
print "%d calls in %d seconds, %.2fms/call" % (count, total_time, ms )
